app/services/projects/bulk_tagger.rb

```diff
@@ -0,0 +1,37 @@
+module Projects
+  class BulkTagger
+    def initialize(content_items:, taxons:)
+      @content_ids = content_items
+      @taxon_content_ids = taxons
+    end
+
+    def commit
+      @content_ids.each do |id|
```

Comment:
this has changed since you last reviewed it. The content_id lookup takes place at the time that a project is created, for one. Also now the actual tagging is handled by a SideKiq worker which gives us free retries in the event of an HTTP error.